framework/Models/Code.py

```diff
@@ -134,26 +137,32 @@ def _readMoreXML(self,xmlNode):
       if child.getName() =='preexec':
         self.preExec = child.value
       elif child.getName() == 'clargs':
-        argtype = child.parameterValues['type']      if 'type'      in child.parameterValues else None
-        arg     = child.parameterValues['arg']       if 'arg'       in child.parameterValues else None
-        ext     = child.parameterValues['extension'] if 'extension' in child.parameterValues else None
+        argtype    = child.parameterValues['type']      if 'type'      in child.parameterValues else None
+        arg        = child.parameterValues['arg']       if 'arg'       in child.parameterValues else None
+        ext        = child.parameterValues['extension'] if 'extension' in child.parameterValues else None
+        # The default delimiter is one empty space
+        delimiter  = child.parameterValues['delimiter'] if 'delimiter' in child.parameterValues else ' '
         if argtype == None:
           self.raiseAnError(IOError,'"type" for clarg not specified!')
         elif argtype == 'text':
           if ext != None:
             self.raiseAWarning('"text" nodes only accept "type" and "arg" attributes! Ignoring "extension"...')
+          if delimiter != " ":
+            self.raiseAWarning('"text" nodes only accept "type" and "arg" attributes! Ignoring "delimiter"...')
           if arg == None:
             self.raiseAnError(IOError,'"arg" for clarg '+argtype+' not specified! Enter text to be used.')
           self.clargs['text']=arg
         elif argtype == 'input':
           if ext == None:
             self.raiseAnError(IOError,'"extension" for clarg '+argtype+' not specified! Enter filetype to be listed for this flag.')
           if arg == None:
-            self.clargs['input']['noarg'].append(ext)
+            self.clargs['input']['noarg'].append((ext,delimiter))
           else:
             if arg not in self.clargs['input'].keys():
               self.clargs['input'][arg]=[]
-            self.clargs['input'][arg].append(ext)
+            # add delimiter to the 'arg' for SAPHIRE interface, since one need to provide
```

Comment:
This comment is misplaced... The Code is not aware of any specific interface (except the RAVEN one) ... generalize the comment

Reply:
changed.